libsyclinterface/source/dpctl_sycl_queue_interface.cpp

```diff
@@ -38,6 +38,100 @@
+        {                                                                      \
+            auto la = local_accessor<unsigned int, NDIM>(R, CGH);              \
+            CGH.set_arg(IDX, la);                                              \
+            return true;                                                       \
+        }                                                                      \
+        case DPCTL_UNSIGNED_INT8:                                              \
+        {                                                                      \
+            auto la = local_accessor<uint8_t, NDIM>(R, CGH);                   \
+            CGH.set_arg(IDX, la);                                              \
+            return true;                                                       \
+        }                                                                      \
+        case DPCTL_LONG:                                                       \
+        {                                                                      \
+            auto la = local_accessor<long, NDIM>(R, CGH);                      \
+            CGH.set_arg(IDX, la);                                              \
+            return true;                                                       \
+        }                                                                      \
+        case DPCTL_UNSIGNED_LONG:                                              \
+        {                                                                      \
+            auto la = local_accessor<unsigned long, NDIM>(R, CGH);             \
+            CGH.set_arg(IDX, la);                                              \
+            return true;                                                       \
+        }                                                                      \
+        case DPCTL_LONG_LONG:                                                  \
+        {                                                                      \
+            auto la = local_accessor<long long, NDIM>(R, CGH);                 \
+            CGH.set_arg(IDX, la);                                              \
+            return true;                                                       \
+        }                                                                      \
+        case DPCTL_UNSIGNED_LONG_LONG:                                         \
+        {                                                                      \
+            auto la = local_accessor<unsigned long long, NDIM>(R, CGH);        \
+            CGH.set_arg(IDX, la);                                              \
+            return true;                                                       \
+        }                                                                      \
+        case DPCTL_SIZE_T:                                                     \
```

Comment:
`size_t` is an alias for `unsigned long long`, so the same comment applies here.

Reply:
In practice maybe, but not in theory. `size_t` equates to `SIZE_MAX` that is the largest size object on a given architecture. The only restriction is it should be at least 16 bit wide, its max is platform dependent. C99's `unsigned long long` is guaranteed to be at least 64 bit wide.

Reply:
@oleksandr-pavlyk would it make sense to switch over from the C99 types to the C++11 types only? That remove all support for `long`, `unsigned long` _etc._ and only support `[u]int*_t` types? 

As the only usage is from numba-dpex, there should not be any other impact. In numba-dpex the logic for supported kernel argument types is as follows:

```python
if ty == types.int32 or isinstance(ty, types.scalars.IntegerLiteral):
        # DPCTL_LONG_LONG
        val = context.get_constant(types.int32, 9)
    elif ty == types.uint32:
        # DPCTL_UNSIGNED_LONG_LONG
        val = context.get_constant(types.int32, 10)
    elif ty == types.boolean:
        # DPCTL_UNSIGNED_INT
        val = context.get_constant(types.int32, 5)
    elif ty == types.int64:
        # DPCTL_LONG_LONG
        val = context.get_constant(types.int32, 9)
    elif ty == types.uint64:
        # DPCTL_SIZE_T
        val = context.get_constant(types.int32, 11)
    elif ty == types.float32:
        # DPCTL_FLOAT
        val = context.get_constant(types.int32, 12)
    elif ty == types.float64:
        # DPCTL_DOUBLE
        val = context.get_constant(types.int32, 13)
    elif ty == types.voidptr or isinstance(ty, types.CPointer):
        # DPCTL_VOID_PTR
        val = context.get_constant(types.int32, 15)
    else:
        raise NotImplementedError
```